prepline_general/api/models/form_params.py

```diff
@@ -0,0 +1,209 @@
+from typing import Annotated, Optional, List, Literal
+
+from fastapi import Form
+
+from pydantic import BaseModel, BeforeValidator
+
+from prepline_general.api.utils import SmartValueParser
+
+
+class GeneralFormParams(BaseModel):
+    xml_keep_tags: bool
+    languages: Optional[List[str]]
+    ocr_languages: Optional[List[str]]
+    skip_infer_table_types: Optional[List[str]]
+    gz_uncompressed_content_type: Optional[str]
+    output_format: str
+    coordinates: bool
+    encoding: str
+    hi_res_model_name: Optional[str]
+    include_page_breaks: bool
+    pdf_infer_table_structure: bool
+    strategy: str
+    extract_image_block_types: Optional[List[str]]
+    chunking_strategy: Optional[str]
+    combine_under_n_chars: Optional[int]
+    max_characters: int
+    multipage_sections: bool
+    new_after_n_chars: Optional[int]
+
+    @classmethod
```

Comment:
I take it this is for the backwards compatibility? Can we have a comment here to clarify how new params should be added?

Reply:
for backward compatibility I added:
```
BeforeValidator(SmartValueParser[List[str]]().value_or_first_element),
```
In old fields that expected values both as a single value or as an element of a list.
The `SmartValueParser` will extract the first element from the list if the expected type is not a container.
`SmartValueParser` is a generic class and is parametrized with the expected type - in the above example its `List[str]`
You can check its tests to see how it's working:
[SmartValueParser tests](https://github.com/Unstructured-IO/unstructured-api/pull/359/files#diff-9c7f757ce9b6069a7eb77b85a1155564bb67e822d2fccd54e383c1714429c755)
[Deprecated API tests (with params passed as lists)](https://github.com/Unstructured-IO/unstructured-api/pull/359/files#diff-465e31562cbb79b8c3177e91026b46eb3f2edc6711d48f462e1f470f528856c0)